lib/number.js

```diff
@@ -53,12 +53,29 @@ internals.Number = class extends Any {
 
     multiple(base) {
 
-        Hoek.assert(Hoek.isInteger(base), 'multiple must be an integer');
```

Comment:
The developer is using joi to validate external inputs, not to validate its own inputs, there are synchronous throws for that yes.
How would allowing floats crash anything ?